### PR TITLE
Make AZ_RETURN_IF... internal.

### LINKED ISSUE #1103: Make AZ_RETURN_IF_FAILED and AZ_RETURN_IF_NOT_ENOUGH_SIZE macros internal
Though these macros are quite useful and used frequently within the SDK, they are currently not intended for consumers of the SDK to use.

Rename them using `_` to indicate they are internal only:
`AZ_RETURN_IF_FAILED` -> `_az_RETURN_IF_FAILED`
`AZ_RETURN_IF_NOT_ENOUGH_SIZE` -> `_az_RETURN_IF_NOT_ENOUGH_SIZE `

https://github.com/Azure/azure-sdk-for-c/blob/9ce11465d167e512e1abdaa912bc0e5c5dec3346/sdk/inc/azure/core/az_result.h#L44-L61

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #1103.

### CODE REVIEW

File: sdk/samples/iot/iot_sample_common.c [hunk 1]
```diff
@@ -82,7 +82,7 @@ static az_result read_configuration_entry(
     (void)printf("%s = %s\n", env_name, hide_value ? "***" : env_value);
     az_span env_span = az_span_create_from_str(env_value);
 
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(destination, az_span_size(env_span));
+    IOT_SAMPLE_RETURN_IF_NOT_ENOUGH_SIZE(destination, az_span_size(env_span));
```
Comment (RickWinter) [inline sdk/samples/iot/iot_sample_common.c:85]: Make the names friendlier.  Remove the `IOT_SAMPLE` portion of the macros `RETURN_IF_NOT_ENOUGH_SIZE`.  If customers copy the code and use it they may also want the MACROS in the sample.

File: sdk/samples/iot/iot_sample_common.h [hunk 2]
```diff
@@ -18,42 +18,65 @@
 //
 // Logging
 //
-#define IOT_SAMPLE_LOG_ERROR(...)                                                  \
-  do                                                                               \
-  {                                                                                \
+#define IOT_SAMPLE_LOG_ERROR(...) \
```
Comment (RickWinter) [inline sdk/samples/iot/iot_sample_common.h:21]: See my previous comment on the naming of these macros

File: sdk/src/azure/platform/az_curl.c [hunk 3]
```diff
@@ -58,7 +59,8 @@ static AZ_NODISCARD az_result _az_http_client_curl_code_to_result(CURLcode code)
 }
 
 // returning AZ error on CURL Error
-#define AZ_RETURN_IF_CURL_FAILED(exp) AZ_RETURN_IF_FAILED(_az_http_client_curl_code_to_result(exp))
+#define _az_RETURN_IF_CURL_FAILED(exp) \
```
Comment (ahsonkhan) [inline sdk/src/azure/platform/az_curl.c:62]: Question: Should this be in a do/while loop?

File: sdk/samples/iot/iot_sample_common.h [hunk 4]
```diff
@@ -137,7 +160,8 @@ typedef enum
  * @return An #az_result value indicating the result of the operation.
  * @retval #AZ_OK All required environment variables successfully read-in.
  * @retval #AZ_ERROR_ARG Sample type or name is undefined, or environment variable is not set.
- * @retval #AZ_RETURN_IF_NOT_ENOUGH_SIZE Not enough space set aside to store environment variable.
+ * @retval #AZ_ERROR_INSUFFICIENT_SPAN_SIZE Not enough space set aside to store environment
```
Comment (ahsonkhan) [inline sdk/samples/iot/iot_sample_common.h:163]: Good catch!

File: sdk/samples/iot/sample_pnp_device_info_component.c [hunk 5]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
+#include "iot_sample_common.h"
```
Comment (ahsonkhan) [inline sdk/samples/iot/sample_pnp_device_info_component.c:4]: Does the ordering of this include matter compared to the others? That is, does it have to be first? Is that why you have a separate group for it?

File: sdk/samples/iot/sample_pnp_thermostat_component.h [hunk 6]
```diff
@@ -9,6 +9,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include <azure/core/az_json.h>
```
Comment (ahsonkhan) [inline sdk/samples/iot/sample_pnp_thermostat_component.h:12]: Lol, why does this keep getting added? I am fine with it, but just curious :)

File: sdk/samples/iot/iot_sample_common.c (hunk 1 above)
Comment (antkmsft) [reply sdk/samples/iot/iot_sample_common.c:85]: @momuno - you came up with this naming prefix recently. Should I keep `IOT_SAMPLE`, or make it something else?

File: sdk/samples/iot/iot_sample_common.h (hunk 2 above)
Comment (antkmsft) [reply sdk/samples/iot/iot_sample_common.h:21]: @momuno - you came up with this naming prefix recently. Should I keep `IOT_SAMPLE`, or make it something else?

File: sdk/src/azure/platform/az_curl.c (hunk 3 above)
Comment (antkmsft) [reply sdk/src/azure/platform/az_curl.c:62]: No. It is the multi-statement macros "returning void" that should be. `AZ_RETURN_IF_FAILED()` would wrap it in do-while.

File: sdk/samples/iot/sample_pnp_device_info_component.c (hunk 5 above)
Comment (antkmsft) [reply sdk/samples/iot/sample_pnp_device_info_component.c:4]: I've seen in many other samples, this include file going first as a group, so this is matching. Plus, in my case, there seems to be a bug in VS formatter - it puts this file non-alphabetically when inside a group (between `s` and `s` - impossible).

File: sdk/samples/iot/sample_pnp_thermostat_component.h (hunk 6 above)
Comment (antkmsft) [reply sdk/samples/iot/sample_pnp_thermostat_component.h:12]: It's really, the compilation fails without it. I don't know why id doesn't always. Include file rearrangement has something to do with it? Include file rearrangement should not normally break compilation, as the inclusion should take care of including everything it needs. And if you look in this file, it uses json types, but nowhere in the chain of include in this file does it include json.h. So it must be that the files using this header just always happened to have json.h included before. So, this is an improvement.

File: sdk/samples/iot/iot_sample_common.c (hunk 1 above)
Comment (antkmsft) [reply sdk/samples/iot/iot_sample_common.c:85]: I contacted Mollie, we will likely follow up outside of this PR.

File: sdk/samples/iot/iot_sample_common.h (hunk 2 above)
Comment (antkmsft) [reply sdk/samples/iot/iot_sample_common.h:21]: I contacted Mollie, we will likely follow up outside of this PR.

File: sdk/samples/iot/sample_pnp_thermostat_component.h (hunk 6 above)
Comment (ahsonkhan) [reply sdk/samples/iot/sample_pnp_thermostat_component.h:12]: Agreed.

And I think you are right. The include reordering is probably breaking the build in some cases.

File: sdk/samples/iot/iot_sample_common.c (hunk 1 above)
Comment (RickWinter) [reply sdk/samples/iot/iot_sample_common.c:85]: I suggest the naming as when code is copied it makes sense for the user w/o needing to rename/replace the MACRO name. 
 IoT can make the final decision on that.

Comment (momuno) [reply sdk/samples/iot/iot_sample_common.c:85]: I chose to go with IOT_SAMPLE on macros to follow the iot_sample prefix used on everything else in the iot_sample_commong.c and .h files.  There are many places that AZ_ prefixed functions are called in the sample, such as with json parsing.  Why does this MACRO name need to change?

File: sdk/samples/iot/iot_sample_common.h (hunk 2 above)
Comment (momuno) [reply sdk/samples/iot/iot_sample_common.h:21]: See my earlier note.  These are used across samples based on the iot_sample_common files and meant to be distinguished from static functions belonging to each sample.  Additionally, it is meant to show that these are custom functions found in io_sample_common and not part of the AZ API.

File: sdk/samples/iot/sample_pnp_thermostat_component.h (hunk 6 above)
Comment (momuno) [reply sdk/samples/iot/sample_pnp_thermostat_component.h:12]: I will look at the re-ording when I get to this sample for finalizing.  Thanks for finding this!

File: sdk/samples/iot/iot_sample_common.c (hunk 1 above)
Comment (ahsonkhan) [reply sdk/samples/iot/iot_sample_common.c:85]: > Why does this MACRO name need to change?

We made the `AZ_RETURN_IF_NOT_ENOUGH_SIZE` macro internal. So, similar to `IOT_SAMPLE_PRECONDITION_NOT_NULL`, the samples need to define their own macro.

**Pull request merged**